### PR TITLE
Change pointers to vectors

### PR DESCRIPTION
**Pull request by DylanWRh**
Change some pointers to vectors in cubic_spline.cpp, math_ylmreal.cpp and opt_CG.cpp

### CODE REVIEW

File: source/module_base/test/formatter_contextfmt_test.cpp
Comment (kirk0830) [inline source/module_base/test/formatter_contextfmt_test.cpp:1]: No, please do not change this function. Try to look at the name of this test case: `StreamOperatorLeft2PtrInt`, it is, test of stream operator left (<<) to pointer/array of int

**Review by kirk0830 — APPROVED** [review]

File: source/module_base/test/formatter_contextfmt_test.cpp
Comment (DylanWRh) [reply source/module_base/test/formatter_contextfmt_test.cpp:1]: Alright. I'll try to find out other pointers that could be replaced by vectors. Thanks for the tip!

Comment (DylanWRh) [reply source/module_base/test/formatter_contextfmt_test.cpp:1]: > No, please do not change this function. Try to look at the name of this test case: `StreamOperatorLeft2PtrInt`, it is, test of stream operator left (<<) to pointer/array of int

Alright. I'll try to find out other pointers that could be replaced by vectors. Thanks for the tip!

File: source/module_base/cubic_spline.cpp [hunk 1]
```diff
@@ -124,13 +125,18 @@ void CubicSpline::build(const int n,
     }
     else
     {
-        double* dx = new double[n - 1];
```
Comment (kirk0830) [inline source/module_base/cubic_spline.cpp:127]: I remember you used to refactor this part, I wonder way you did not use std::vector then? Do you have some special consideration? About heterogeneous calculation? @jinzx10

**Review by kirk0830 — CHANGES_REQUESTED** [review]
cubsplin is the part used to be refactoed by @jinzx10, ask for comments from him would be safe and beneficial.

**Review by jinzx10 — APPROVED** [review]

**Review by kirk0830 — APPROVED** [review]

**Review by mohanchen — APPROVED** [review]

**Pull request merged**